### PR TITLE
Capture decisions made about designs.

### PR DESCRIPTION
**Pull request by paddycarver**
Add recommendations and the justifications we made when deciding things to the design documents we've made decisions about.

### CODE REVIEW

File: docs/design/factories.md [hunk 1]
```diff
@@ -432,6 +432,36 @@ however, this avenue will no longer be available, and we'll either need to
...
 decide that provider developers don't need access to it.
 
+## Recommendations
+
+We believe the reflection and manually copying options are dangerous and are
+strongly recommending against them. They're not idiomatic and they hide too
+much of what's happening.
+
+Named functions and anonymous functions are mostly useful on structs, and we're
+shying away from that pattern, so we're recommending against them.
```
Comment (kmoe) [inline docs/design/factories.md:442]: Perhaps link to the structs/interfaces doc here

**Review by kmoe — APPROVED** [review]
Excellent summaries

File: docs/design/factories.md (hunk 1 above)
Comment (paddycarver) [reply docs/design/factories.md]: Good call. :+1:

**Review by bflad — APPROVED** [review]
Looks good to me 🚀

File: docs/design/structs-interfaces.md [hunk 2]
```diff
@@ -302,3 +302,34 @@ type ResourceWithContext interface {
...
+real danger, and something we should keep an eye on mitigating as we implement.
+A clear and obvious tool, though not by any means a complete solution, is
+documenting what is expected and appropriate, and what is unexpected and
+unsupported in the GoDoc comments for the places where users have the ability
+to create issues for themselves.
+
+We feel that structs are the most obvious entrypoint, and offer the best
+first-five-minutes experience to provider developers, because they're filling
+in blanks instead of defining their own types. However, we feel that interfaces
+offer the best first-hour experience to provider developers, because of the
```
Comment (aareet) [inline docs/design/structs-interfaces.md:326]: Should we include that we discussed using provider scaffolding to reduce this time-to-productivity for first time provider developers?

**Review by aareet — APPROVED** [review]
Just the one comment. Looks great!

**Pull request merged**